io/src/main/java/io/smallrye/common/io/PathUtils.java

```diff
@@ -0,0 +1,2139 @@
+package io.smallrye.common.io;
+
+import static io.smallrye.common.constraint.Assert.checkNotEmptyAfterTrimParam;
+import static io.smallrye.common.constraint.Assert.checkNotNullParam;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.AccessMode;
+import java.nio.file.CopyOption;
+import java.nio.file.DirectoryIteratorException;
+import java.nio.file.DirectoryNotEmptyException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.FileSystemException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.NotDirectoryException;
+import java.nio.file.NotLinkException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SecureDirectoryStream;
+import java.nio.file.attribute.BasicFileAttributeView;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.FileTime;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import io.smallrye.common.constraint.Assert;
+
+/**
+ * Utilities for {@link java.nio.file.Files}. Fluent style methods, more input
+ * types and unchecked exceptions make these more convenient.
+ *
+ * @author Boris Unckel
+ *
+ */
+public final class PathUtils {
+
+    public static final LinkOption[] NO_FOLLOW_LINK_OPTION_ARRAY = { LinkOption.NOFOLLOW_LINKS };
+
+    /**
+     * A wrapper around {@link java.nio.file.Paths#get(String, String...)} with
+     * explicit null check.
+     *
+     * @param first part of the path, must be not null, not empty and a valid path
+     * @param more optional parts of the path
+     * @return the path constructed out of first and more
+     * @throws IllegalArgumentException if first is null or empty
+     */
+    public static Path get(final String first, final String... more) {
+        return Paths.get(checkNotEmptyAfterTrimParam("first", first), more);
+    }
+
+    /**
+     * A wrapper around {@link java.nio.file.Paths#get(URI)} with explicit null
+     * check.
+     *
+     * @param uri for the path, must be not null and a valid path
+     * @return the path constructed out of the uri
+     * @throws IllegalArgumentException if uri is null
+     */
+    public static Path get(final URI uri) {
+        return Paths.get(checkNotNullParam("uri", uri));
```

Comment:
I believe that `Paths.get()` also does a `null` check; why wrap it?

Reply:
I have checked that and could not find a explicit null check. The intention is to have a Exception with the parameter name. I don't have a personal favour for NullPointer vs. IllegalArgument but I dislike the anonymous, fail late ones.